From decdb9a4bc0f050cca2d9d92525559e041527a4a Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Tue, 20 Jun 2023 14:36:53 +0200
Subject: [PATCH] ARM: dts: stm32: add TAMP backup register on STM32MP13

TAMP backup registers allow information sharing between different
software components/execution levels.
They are described individually as nvmem cells.

Change-Id: Iea49ac0fcb69acc82a70a1cdfeb4717bcec6da5e
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -1630,6 +1630,34 @@
 			};
 		};
 
+		tamp: tamp@5c00a000 {
+			compatible = "st,stm32-tamp", "syscon", "simple-mfd";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x5c00a000 0x400>;
+			ranges;
+
+			nvram: nvram@5c00a100 {
+				compatible = "st,stm32mp15-tamp-nvram";
+				#address-cells = <1>;
+				#size-cells = <1>;
+				reg = <0x5c00a100 0x80>;
+
+				saes_secret_key: tamp-bkp@0 {
+					/* see saes secret key feature */
+					reg = <0x0 0x20>;
+				};
+				boot_mode: tamp-bkp@78 {
+					/* see boot mode selection feature */
+					reg = <0x78 0x4>;
+				};
+				boot_counter: tamp-bkp@7c {
+					/* see boot counter feature */
+					reg = <0x7c 0x4>;
+				};
+			};
+		};
+
 		/*
 		 * Break node order to solve dependency probe issue between
 		 * pinctrl and exti.
